src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -26,12 +33,68 @@ public void infix(Section infixSection) {
         add(infixSection);
     }
 
+    public void delete(Station station) {
+        if (isHaveOnlyOneSection()) {
+            throw new InvalidParameterException("단일 구간인 노선은 역을 제거할 수 없습니다.");
+        }
+        List<Section> sectionsContainStation = getSectionsContainStation(station);
+        deleteStation(sectionsContainStation, station);
+    }
+
+    private boolean isHaveOnlyOneSection() {
+        if (sections.size() < 2) {
+            return true;
+        }
+        return false;
+    }
+
+    private void deleteStation(List<Section> sectionsContainStation, Station station) {
+        if (sectionsContainStation.size() == 0) {
+            throw new InvalidParameterException("호선에 존재하지 않는 역은 제거할 수 없습니다.");
+        }
+
+        if (sectionsContainStation.size() == 2) {
+            sections.add(createDeleteSection(sectionsContainStation, station));
+            sections.remove(sectionsContainStation.get(1));
+        }
+        sections.remove(sectionsContainStation.get(0));
+    }
+
+    private Section createDeleteSection(List<Section> sectionsContainStation, Station station) {
+        return sectionsContainStation.get(0).delete(sectionsContainStation.get(1), station);
+    }
+
+    private List<Section> getSectionsContainStation(Station station) {
+        List<Section> sectionsContainStation = new ArrayList<>();
+        for (Section section : sections) {
+            sectionsContainStation.add(getSectionContainStation(section, station));
+            sectionsContainStation.remove(null);
+        }
+        return sectionsContainStation;
+    }
+
+    private Section getSectionContainStation(Section section, Station station) {
+        if (section.getUpStation().equalsById(station)) {
+            return section;
+        }
+        if (section.getDownStation().equalsById(station)) {
+            return section;
+        }
+        return null;
+    }
```

Comment:
```suggestion
    private List<Section> getSectionsContainStation(Station station) {
        return sections.stream()
                .filter(section -> section.contains(station))
                .collect(Collectors.toList());
    }
```
null을 `add`했다가 `remove`하기보다는
이렇게 작성해보는건 어떨까요?
`section.getUpstation()` 과 같은 getter도 제거할수 있고 로직도 파악하기 쉬운것 같은데 어떻게 생각하시나요?

Reply:
와ㅏ아아아아 이렇게도 구현해도 되겠네요 ㅠㅠ
훨씬 코드도 깔끔하고 section 내의 필드에 접근하는 행위를 section 도메인에서 구현할수 있어서
굉장히 좋은거 같아요!!!!!

Reply:
와 다시보니 정말 stream이 컴팩트하고 좋네요
sections 를 순회하면서 무언가의 조건에 충족된 데이터를 모아 리스트로 구현하는 행위하는 로직을

1. sections 를 순회하면서 `stream`
2. 무언가의 조건에 충족된 데이터를 모아 `filter`
3. 리스트로 구현 `collect`

하게 깔끔하게 구성이 되는.......👍🏻